src/els_execute_command_provider.erl

```diff
@@ -0,0 +1,50 @@
+-module(els_execute_command_provider).
+
+-behaviour(els_provider).
+
+-export([ handle_request/2
+        , is_enabled/0
+        , options/0
+        ]).
+
+-include("erlang_ls.hrl").
+
+%%==============================================================================
+%% els_provider functions
+%%==============================================================================
+
+-spec is_enabled() -> boolean().
+is_enabled() -> true.
+
+-spec options() -> map().
+options() ->
+  #{ commands => ["replace-lines"] }.
+
+-spec handle_request(any(), els_provider:state()) ->
+  {any(), els_provider:state()}.
+handle_request({workspace_executecommand, Params}, State) ->
+  #{ <<"command">> := Command } = Params,
+  Arguments = maps:get(<<"arguments">>, Params, []),
+  Result = execute_command(Command, Arguments),
+  {Result, State}.
+
+%%==============================================================================
+%% Internal Functions
+%%==============================================================================
+
+-spec execute_command(string(), [any()]) -> [map()].
+execute_command(<<"replace-lines">>
```

Comment:
@alanz I tried this locally, but I could not trigger it from Emacs. How should I test this?

Reply:
It gets triggered via the code action. That is why this thing has to come in as a whole.   The process in general is

1.  Something generates diagnostics
2. The client requests code actions based on the current location in the file and the diagnostics that are active for that location.
3. Each code action provider examines the diagnostics (or other info) and generates a list of possible code actions. These can either be literal, where there is an embedded `ApplyEdit`, or contain a `command` and its parameters.
4. When the user selects a code action, if it has a literal edit, this is applied. Otherwise the `command` and it's parameters are send in an `executeCommand` request.
5. The response to the `executeCommand` is normally just empty, to indicate all went well.
6. The server then sends a `workspace/applyEdit` command to the client, with the changes derived from executing the command.